test/code/providers/Container_ImageInventory_Class_Provider_UnitTest.cpp

```diff
@@ -53,6 +53,18 @@ class ImageInventoryTest : public CppUnit::TestFixture
 		return processOutput.str();
 	}
 
+	// extract substring and trim trailing space or tab
+	static wstring substrTrim(wstring line, string::size_type begin, string::size_type end)
+	{
+		wstring sub = line.substr(begin, end - begin);
+		string::size_type trim = sub.find_last_not_of(L" \t");
+		if (trim != string::npos)
+		{
+			return sub.substr(0, trim + 1);
+		}
+		return sub;
+	}
+
```

Comment:
There's a stringutils thing for that, see SCXCoreLIb::StrTrim in file `pal/source/code/include/scxcorelib/stringaid.h`. To avoid code duplication needlessly, considering that this project already requires the PAL, I'm going to ask you to make that change before accepting this.